src/daemon/daemon.cpp

```diff
@@ -114,21 +114,38 @@ mp::Query query_from(const mp::LaunchRequest* request, const std::string& name)
     return {name, image, false, request->remote_name(), query_type, true};
 }
 
-auto make_cloud_init_vendor_config(const mp::SSHKeyProvider& key_provider, const std::string& time_zone,
-                                   const std::string& username, const std::string& backend_version_string)
+auto make_cloud_init_vendor_config(const mp::SSHKeyProvider& key_provider, const std::string& username,
+                                   const std::string& backend_version_string, const mp::CreateRequest* request)
 {
     auto ssh_key_line = fmt::format("ssh-rsa {} {}@localhost", key_provider.public_key_as_base64(), username);
+    QString pollinate_alias = QString::fromStdString(request->image());
 
+    if (pollinate_alias.isEmpty())
+    {
+        pollinate_alias = "default";
+    }
+    else if (pollinate_alias.startsWith("http"))
+    {
+        pollinate_alias = "http";
+    }
+    else if (pollinate_alias.startsWith("file"))
+    {
+        pollinate_alias = "file";
+    }
+
+    auto remote_name = request->remote_name();
     auto config = YAML::Load(mp::base_cloud_init_config);
     config["ssh_authorized_keys"].push_back(ssh_key_line);
-    config["timezone"] = time_zone;
+    config["timezone"] = request->time_zone();
     config["system_info"]["default_user"]["name"] = username;
 
     auto pollinate_user_agent_string =
         fmt::format("multipass/version/{} # written by Multipass\n", multipass::version_string);
     pollinate_user_agent_string += fmt::format("multipass/driver/{} # written by Multipass\n", backend_version_string);
     pollinate_user_agent_string +=
         fmt::format("multipass/host/{} # written by Multipass\n", multipass::platform::host_version());
+    pollinate_user_agent_string += fmt::format("multipass/alias/{}{} # written by Multipass\n",
+                                               !remote_name.empty() ? remote_name + ":" : "", pollinate_alias);
```

Comment:
Hey @nathan-j-hart, this is capturing the `remote` (if provided by the user) and the "alias" used to `launch` the instance.  Is that good for what you need?

Reply:
where `remote` is whether it's a URL or file? alias is the biggest need here so looks good from that standpoint

Reply:
`remote` (if provided) will tell us if it's say, for example, a `snapcraft:core18` launch (which is a buildd image for building snaps) vs. a `core18` launch (which is the actual Ubuntu Core 18 distro).  This helps discern this ambiguity.

Reply:
Also, `alias` will either be `http` or `file` only if it's those types of launches.  No other part of the url will be part of the string.

Reply:
Right, ok in that case this looks great to me